soweego/linker/train.py

```diff
@@ -137,12 +162,15 @@ def _build_positive_samples_index(wd_reader1):
     return positive_samples_index
 
 
-def _train(classifier, feature_vectors, positive_samples_index, binarize):
-
-    model = workflow.init_model(classifier, binarize, feature_vectors.shape[1])
+def _train(classifier, feature_vectors, positive_samples_index, **kwargs):
+    model = _initialize(classifier, feature_vectors, **kwargs)
 
     LOGGER.info('Training a %s', classifier)
     model.fit(feature_vectors, positive_samples_index)
 
     LOGGER.info('Training done')
     return model
+
+
+def _initialize(classifier, feature_vectors, **kwargs):
+    return workflow.init_model(classifier, feature_vectors.shape[1], **kwargs) if classifier is constants.SINGLE_LAYER_PERCEPTRON else workflow.init_model(classifier, **kwargs)
```

Comment:
If we're going to leave this `_initialize` then: Also here, after merging with master the `mlp` classifier should also be considered 

```suggestion
    return workflow.init_model(classifier, feature_vectors.shape[1], **kwargs) if classifier in (constants.SINGLE_LAYER_PERCEPTRON, constants.MULTILAYER_CLASSIFIER) else workflow.init_model(classifier, **kwargs)
```